### ISSUE TITLE
Trials with same hyperparameters - Random search 

### ISSUE THREAD

**Issue by midhun1998** [kubeflow/katib#1566]
/kind bug

**What steps did you take and what happened:**
I had recently run 8 trials for an experiment and out of 8 only 4 unique combinations were seen and the rest were duplicates. The duplicates hyperparameters are seen in the majority of the experiments I run.  
Ref:
#842 

Looking forward to this feature. #866 

**What did you expect to happen:**
Expected 8 different combinations of hyperparameter to run.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
Suggestion pod log:
```
INFO:pkg.suggestion.v1beta1.hyperopt.base_service:GetSuggestions returns 8 new Trial
```

Suggestion controller log: (I might have missed a few as many different experiments ran after this)
```
{"level":"info","ts":1624915551.6826324,"logger":"suggestion-controller","msg":"Sync assignments","Suggestion":"admin/test-123","Suggestion Requests":8,"Suggestion Count":8}
{"level":"info","ts":1624925599.2925694,"logger":"suggestion-controller","msg":"Sync assignments","Suggestion":"admin/test-123","Suggestion Requests":8,"Suggestion Count":8}
{"level":"info","ts":1624929816.4245157,"logger":"suggestion-controller","msg":"Sync assignments","Suggestion":"admin/test-123","Suggestion Requests":8,"Suggestion Count":8}
{"level":"info","ts":1624953809.892868,"logger":"suggestion-controller","msg":"Sync assignments","Suggestion":"admin/test-123","Suggestion Requests":8,"Suggestion Count":8}
```
Search space:
```
lr: 0.001
epochs: 100
batch size: 32, 64
num of cells: 40, 60
num of layers: 2, 3
```
Trail configuration:
```
Parallel trials: 8
Max trials: 8
Max failed trails: 2
```
Result:
![123675979-f55e7380-d860-11eb-8e49-5d3d402e8aa6](https://user-images.githubusercontent.com/24776450/123824409-a590b280-d91b-11eb-8071-bafa2028a195.jpg)

This is all I can add as of now as the pods were cleaned up. If this isn't enough please let me know what all I need to record when this happens again. 

**Environment:**

- Kubeflow version (`kfctl version`): 1.3
- Kubernetes version: (use `kubectl version`): 1.19
- Katib controller version: v0.11.0

**Comment by johnugeorge** [comment]
Thanks @midhun1998 

you have exactly 8 possible configurations from your search space and your max trials is also 8. Is it right? Can you reproduce this issue each time?

**Comment by RDarrylR** [comment]
I have seen the same in the past with kf1.2. I will try again with 1.3 and report results. I also saw the same with grid algorithm before which kind of defeats the purpose of using grid. I will test with 1.3.

**Comment by johnugeorge** [comment]
https://github.com/kubeflow/katib/issues/1534 might be a related issue.

**Comment by midhun1998** [comment]
Yeah, @johnugeorge . I tried running it again today twice and I saw duplicates. Here are the details:

First run:  **test-new-npdr7**

Suggestion pod logs: 
```
$ kubectl logs  test-new-npdr7-random-7895f4bc6fjtlm2 
INFO:pkg.suggestion.v1beta1.hyperopt.base_service:GetSuggestions 
returns 8 new Trial
INFO:pkg.suggestion.v1beta1.hyperopt.base_service:GetSuggestions 
returns 8 new Trial
```
Suggestion controller logs:
https://gist.github.com/midhun1998/4a4cb69671f3033fecddd195dc787fcd

UI:
<img width="745" alt="test-new" src="https://user-images.githubusercontent.com/24776450/123950906-d75c5480-d9c1-11eb-9009-d5e27871069e.png">

Second run:  **test-new-npdr7**

Suggestion pod logs: 
```
~ $ kl test-extra-xyf23-random-766dc887668ldv
INFO:pkg.suggestion.v1beta1.hyperopt.base_service:GetSuggestions 
returns 8 new Trial
INFO:pkg.suggestion.v1beta1.hyperopt.base_service:GetSuggestions 
returns 8 new Trial
```
Suggestion controller logs:
https://gist.github.com/midhun1998/d219d95612378ca17eb5fef588099d36

UI:
<img width="704" alt="test-extra" src="https://user-images.githubusercontent.com/24776450/123951320-4a65cb00-d9c2-11eb-9d59-035c563fe41a.png">

Observations:

- Second run has few more unique values than other attempts.
- A weird thing noticed in suggestion controller logs is that 2 different suggestions were given for the same experiment but only one of them was actually taken.

**Comment by johnugeorge** [comment]
I will try to reproduce this one. Thanks for reporting this

**Comment by midhun1998** [comment]
Hi @johnugeorge . Any update on this? Were you able to reproduce?

**Comment by johnugeorge** [comment]
related: #1633
